snuba/split.py

```diff
@@ -9,86 +9,150 @@
 # queries before hitting the 90d limit (2+20+200+2000 hours == 92 days).
 STEP_GROWTH = 10
 
+
 def split_query(query_func):
-    """
-    If a query is:
-        - ORDER BY timestamp DESC
-        - has no grouping
-        - has an offset/limit
-        - has a large time range
-    We know we have to reverse-sort the entire set of rows to return the small
-    chunk at the end of the time range, so optimistically split the time range
-    into smaller increments, and start with the last one, so that we can potentially
-    avoid querying the entire range.
-    """
     def wrapper(*args, **kwargs):
         body = args[0]
-        use_split, date_align, split_step = state.get_configs([
+        use_split = state.get_configs([
             ('use_split', 0),
+        ])
+        limit = body.get('limit', 0)
+        remaining_offset = body.get('offset', 0)
+        orderby = util.to_list(body.get('orderby'))
+
+        total_col_count = len(util.all_referenced_columns(body))
+
+        common_conditions = use_split and limit and not body.get('groupby')
+
+        if common_conditions:
+            min_col_count = len(util.all_referenced_columns(
+                {**body, 'selected_columns': ['project_id', 'event_id', 'timestamp']}))
```

Comment:
Not a big deal, but both `total_col_count` (line 23) and `min_col_count` are going to get eagerly evaluated since they are defined as local variables and not just part of the boolean expression that starts on line 32. (Same with `remaining_offset` on line 20 and `orderby` on line 21, these local variables may be set and never utilized.)